desktop/main.js

```diff
@@ -36,6 +36,31 @@ autoUpdater.logger.transports.file.level = 'info';
 // See https://www.npmjs.com/package/electron-log
 Object.assign(console, log.functions);
 
+// Handle auto-update logic
+const OPEN_HIDDEN_COMMAND_LINE_ARG = '--open-hidden';
+
+let updateDownloaded = false;
+
+const electronUpdater = browserWindow => ({
+    init: () => {
+        autoUpdater.on('update-downloaded', () => {
+            updateDownloaded = true;
+            if (!browserWindow.isVisible()) {
+                autoUpdater.quitAndInstall();
+            }
+        });
+
+        browserWindow.on('hide', () => {
+            if (!browserWindow.isVisible() && updateDownloaded) {
+                autoUpdater.quitAndInstall();
+            }
+        });
```

Comment:
In addition to adding event listeners here, you should do an initial check for updates.